yksgi/src/lib.rs

```diff
@@ -98,6 +104,19 @@ signext!(sext_i8, i8);
 signext!(sext_i16, i16);
 signext!(sext_i32, i32);
 
+unsafe fn get_stackmap_call(pc: Value) -> Value {
```

Comment:
Why is the whole function `unsafe`?

Reply:
Tbh I'm still slightly confused about `unsafe` and where to use it. I think my reasoning was that depending on which value of pc you pass in this might not work. So it's on the caller to make sure the argument is safe.

Reply:
In some sense *any* function won't work if you call it incorrectly! In our context `unsafe` should mostly mean "if you call this with the wrong arguments the program may exhibit undefined behaviour (inc. but not limited to segfaulting)". Is this one of those functions?

Reply:
Not sure about segfaulting. I'm inheriting the unsafe from the llvm-sys calls (e.g. `LLVMGetPreviousInstruction`). Not entirely certain how you can get them to segfault but I imagine handing them a nullptr would do the trick.

So as long as we don't do that we should mark the LLVM-calls unsafe instead of the whole function?

Reply:
The `get_stackmap_call` isn't checking any preconditions. I don't know what might result in a segfault or UB in `LLVMGetPreviousInstruction`, so I wouldn't know what to check for. So maybe marking the function as `unsafe` is fine here?

Reply:
Okay, I had a poke around in the LLVM API sources to see where there is potential for UB. It looks like as long as you pass the API the correct type its expecting things should be fine. So I've added some extra asserts to make sure we are not passing nonsense and removed the unsafe from the function. Does that sound ok? See [5993877](https://github.com/ykjit/yk/pull/578/commits/599387756b3ae04da9bccab4d02c586b72e52a7a)

Reply:
Perfect!